lib/spack/spack/directives.py

```diff
@@ -218,6 +218,8 @@ def _execute(pkg):
         if checksum:
             kwargs['md5'] = checksum
 
+        kwargs['version'] = ver
```

Comment:
Hmm... after 20 minutes of poking around the code, `FetchStrategy` is mysterious to me.  I haven't been able to answer basic questions like:

1. Where are different `FetchStrategy`s instantiated?  I can see `URLFetchStrategy` in `from_url()`.  But a grep for `GitFetchStrategy` yields nothing obvious.  Is the code being super-clever somewhere and turning the string `"git"` into the class `GitFetchStrategy`?

2. What is the API for `FetchStrategy`?  Clearly, the package name and version should be available to fetch strategies in a clean way.  (As for package name... both the Spack package name and the download package name should be available.  They are not always the same.)